lib/ohai/plugins/network.rb

```diff
@@ -159,7 +166,9 @@ def network_contains_address(address_to_match, ipaddress, iface)
           Ohai::Log.debug("unable to detect ip6address")
         else
           ip6address r["ip"]
-          if r["mac"] and macaddress.nil? and ipaddress.nil?
+          # don't overwrite macaddress set by "#{os}::network" plugin
+          # and also
+          if r["mac"] and macaddress.nil? and ( ipaddress.nil? || ipaddress == "127.0.0.1" )
```

Comment:
we should flag this - we have a strong preference for only taking from v6 if v4 isn't around, but that's a bad assumption. People will soon be doing v4 on tunnel interfaces in v6 places... this is fine for now as it preserves the previous behavior, but it probably wont' be good forever.